### PR TITLE
Vector mode aggregates and vectors

### PR DESCRIPTION
**Pull request by tgymnich**

### CODE REVIEW

File: enzyme/Enzyme/AdjointGenerator.h [hunk 1]
```diff
@@ -437,13 +437,15 @@ class AdjointGenerator
     constantval |= gutils->isConstantValue(&I);
 
     BasicBlock *parent = I.getParent();
-    Type *type = I.getType();
+    Type *type = gutils->getWidth() > 1
```
Comment (wsmoses) [inline enzyme/Enzyme/AdjointGenerator.h:440]: getShadowType ?

File: enzyme/Enzyme/AdjointGenerator.h [hunk 2]
```diff
@@ -610,40 +612,49 @@ class AdjointGenerator
...
             if (!mask) {
+
+              auto rule = [&](Value *ip) {
 #if LLVM_VERSION_MAJOR > 7
-              auto LI = Builder2.CreateLoad(
-                  cast<PointerType>(I.getOperand(0)->getType())
-                      ->getElementType(),
-                  gutils->invertPointerM(I.getOperand(0), Builder2));
+                auto LI = Builder2.CreateLoad(
+                    cast<PointerType>(ip->getType())->getElementType(), ip);
```
Comment (wsmoses) [inline enzyme/Enzyme/AdjointGenerator.h:623]: Use I.getType()?

File: enzyme/Enzyme/AdjointGenerator.h (hunk 1 above)
Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h]: done

File: enzyme/Enzyme/AdjointGenerator.h (hunk 2 above)
Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h]: done

**Review by wsmoses — APPROVED** [review]

**Pull request merged**